networks:
  reverse-proxy:
    external: true

services:
  # sudo chown -R 472:472 ${USER_DIR}/grafana
  grafana:
    image: docker.io/grafana/grafana:11.4.0@sha256:d8ea37798ccc41061a62ab080f2676dda6bf7815558499f901bdb0f533a456fb
    container_name: grafana
    cpus: 0.5
    mem_limit: 512m
    restart: unless-stopped
    # user: ${GID}:${UID}
    expose:
      - 3000
    networks:
      - reverse-proxy
    healthcheck:
      test: ["CMD", "curl", "--fail", "http://grafana:3000"]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 10s
      start_interval: 5s
    volumes:
      - ${USER_DIR}/grafana:/var/lib/grafana
    environment:
      - TZ=${TZ:?error}
      - GF_USERS_ALLOW_SIGN_UP=false
    labels:
      - traefik.enable=true
      - traefik.http.routers.grafana.entrypoints=websecure
      - traefik.http.routers.grafana.rule=Host(`grafana.${DOMAIN_NAME}`)
      - traefik.http.routers.grafana.middlewares=chain-authelia@file
      - traefik.http.services.grafana.loadbalancer.server.port=3000
